docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+# Task Resource
+
+This resource will:
+
+1. Create cloud resources (storage, machines) for the task
+2. Upload the given `directory` to the cloud storage, if specified
+3. Run the given `script` upon completion or `timeout` in the cloud
+
+## Example Usage
+
+```hcl
+resource "iterative_task" "task" {
+  name  = "example"
+  cloud = "aws"
+
+  environment = {GREETING = "Hello, world!"}
+  directory   = "${path.root}/shared"
+
+  script = <<-END
+    #!/bin/bash
+    echo "$GREETING" | tee $(uuidgen)
+  END
+}
+```
+
+## Argument Reference
+
+### Required
+
+- `name` - (Required) Task name.
+- `cloud` - (Required) Cloud provider to run the task on; valid values are `aws`, `gcp`, `az` and `k8s`.
```

Comment:
Are we just assuming people know how to pass these arguments? It's not shown in the **Example Usage** block.

Reply:
As far as I know, the `cloud` attribute is part of the **Example Usage** section, but not every attribute has been included in the examples. Would you prefer examples to cover every attribute with example values?

Reply:
It's safe to assume that people knows how to specify them. Still, it would be nice to show examples for every argument as part of the **Getting Started** guide. Adding to https://github.com/iterative/terraform-provider-iterative/pull/237#issue-1048177975...